Code/Framework/AzCore/AzCore/std/string/regex.h

```diff
@@ -1758,7 +1758,7 @@ namespace AZStd
             return (*this);
         }
 
-        ~basic_regex()
+        virtual ~basic_regex()
```

Comment:
basic_regex is overridden somewhere? thats odd

Reply:
It shouldn't be.

Reply:
`basic_regex` inherits publicly from `Internal::ErrorSink` which has a virtual method. By having a virtual method makes the base class to have a vtable and therefore the derived class needs a vtable. When the destructor is not virtual and there is a vtable, the compiler warns since the type can be used polymorphically (and therefore could be deleted polymorphically).

Reply:
So does it need to be override instead of virtual then?

Reply:
I kinda agree with @spham-amzn, the base class `Internal::ErrorSink` should gain a `virtual` destructor, and this one should be `override`.
I think a better fix would be to remove the base class and the virtual method, and find a different way to do that error reporting. But that can be a different change.

Reply:
ok, changing